eth/vm/base.py

```diff
@@ -182,7 +184,8 @@ def apply_transaction(self,
     @staticmethod
     def create_execution_context(header: BlockHeaderAPI,
                                  prev_hashes: Iterable[Hash32],
-                                 chain_context: ChainContextAPI) -> ExecutionContextAPI:
+                                 chain_context: ChainContextAPI,
+                                 import_block: BlockAPI = None) -> ExecutionContextAPI:
```

Comment:
Adding `import_block` here feels a bit weird given that there's already `header` in this API. There's has to be a cleaner way to do this but let me try to elaborate what the problem is:

When we import a block, we need to determine the signer of the block so that we know where to send the tx fees. Holding on to this information is the job of the `ExecutionContextAPI`. However, to retrieve the signer we need to have access to the full header that is being imported.

The problem is that our current flow just cherry picks the attributes from the import block which are needed to satisfy the PoW flow, leaving us with a header that we can not use to retrieve the signer from.

Reply:
See my comment above about confusion between how `import_block.header` and `header` are different....

Reply:
It seem unavoidable to give access to the full block/header that is being imported to support a wide range of use cases (including the one at hand).

I inlined the call to `build_state` so that we can leave that API untouched (instead of just passing the `import_block` through).

I still don't like this part but I'm not sure about the right way to approach this.

Reply:
Is `import_block.header == header`?

Reply:
It's not. The `block` that we import gets dropped in `import_block` and we just copy a few things over into our *in progress* block.

https://github.com/ethereum/py-evm/blob/83d50fa2d9aae931775af7ca2c5bac60ad811566/eth/vm/base.py#L285-L324

However, now that I inlined the call to `self.build_state` it occurred to me that I could use the importing block for the header that we pass to `create_execution_context` but use the *in progress* block to read the `state_root` to build the state from. That allows us to to drop the `import_block` here :tada: